app/src/main/java/com/keylesspalace/tusky/EmojiPreference.java

```diff
@@ -0,0 +1,268 @@
+            adapter.onDownloaded(fontList);
+        }
+        // Configure the RecyclerView
+        fontRecycler.setItemAnimator(new DefaultItemAnimator());
+        fontRecycler.setLayoutManager(new LinearLayoutManager(view.getContext()));
+        fontRecycler.setAdapter(adapter);
+    }
+
+    /**
+     * This method is called when we finished downloading our font list.
+     * @param emojiFont The File containing the JSON file with the font list in it
+     */
+    private void onDownloaded(File emojiFont) {
+        // So we're ready to show the actual content! No more loading screens*! (*almost)
+        for (View view: loading) {
+            view.setVisibility(View.GONE);
+        }
+        for (View view: finished) {
+            view.setVisibility(View.VISIBLE);
+        }
+    }
+
+    /**
+     * This interface is used to let objects get notified when the font list has been downloaded.
+     */
+    public interface EmojiFontListener {
+        void onDownloaded(File emojiFont);
+        /*
+        TODO: Maybe add something if downloading went wrong.
+        The only problem about this would be that Lambda expressions couldn't be used anymore :'(
+        */
+    }
+
+    /**
+     * This task bundles all the stuff required to download a new emoji file list
+     */
+    private static class FontListDownloader extends AsyncTask<File, Void, File> {
+        // All the objects interested about (?) this download
+        private EmojiFontListener[] listeners;
+
+        FontListDownloader(EmojiFontListener... listeners) {
+            super();
+            this.listeners = listeners;
+        }
+
+        @Override
+        protected File doInBackground(File... files){
+            // This is (mostly) shamelessly copied from StackOverflow...
+            // We just want to download to ONE file...
+            File fontList = files[0];
+            // This will be used to write the downloaded content
+            BufferedSink sink;
+            try {
+                // There's probably no file as we first need to download it here!
+                if (!fontList.exists()) {
+                    fontList.createNewFile();
+                }
+                OkHttpClient client = new OkHttpClient();
+                // Build a request to download this list.
+                Request request = new Request.Builder().url(FONTS_URL)
+                        .addHeader("Content-Type", "application/json")
+                        .build();
+                // Start download
+                Response response = client.newCall(request).execute();
+                // Open up the writing part
+                sink = Okio.buffer(Okio.sink(fontList));
+                try {
+                    if (response.body() != null) {
+                        // GOGOGO! DOWNLOAD!
+                        sink.writeAll(response.body().source());
+                    } else {
+                        Log.e("FUCK", "downloadFonts: Source empty");
```

Comment:
lol. Please use a proper logging tag

Reply:
This is something I forgot to remove. I did this to make debugging the app a little bit easier since I could just search the logcat 😅